lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -873,9 +884,9 @@ SILCombiner::propagateConcreteTypeOfInitExistential(FullApplySite AI,
   ArchetypeType *OpenedArchetype = nullptr;
   SILValue OpenedArchetypeDef;
   bool isCopied = false;
-  SILInstruction *InitExistential =
-    findInitExistential(AI, Self, OpenedArchetype, OpenedArchetypeDef,
-                        isCopied);
+  DominanceInfo *DT = DA->get(AI.getFunction());
+  SILInstruction *InitExistential = findInitExistential(
+      AI, Self, OpenedArchetype, OpenedArchetypeDef, isCopied, DT);
```

Comment:
The the dominance test fails, you should not bail out here, but after 'Propagate(ConcreteType, Conformance);'
Even if we can't propagate the value we want to at least propagate the type.

Reply:
I just tried doing that: we get SILVerifier failures if we do so:
```
SIL verification failed: open_existential_addr uses that consumes or mutates but is not opened for mutation: allowedAccessKind == OpenedExistentialAccess::Mutable || !isMutatingOrConsuming(OEI)
Verifying instruction:
     %8 = alloc_stack $Encoder                    // users: %164, %95, %14, %11, %96, %165, %9
->   %14 = open_existential_addr immutable_access %8 : $*Encoder to $*@opened("66D0089C-AAE5-11E7-8FCE-3200190FD320") Encoder // user: %15
     %15 = unchecked_addr_cast %14 : $*@opened("66D0089C-AAE5-11E7-8FCE-3200190FD320") Encoder to $*_JSONEncoder // user: %16
In function:
```